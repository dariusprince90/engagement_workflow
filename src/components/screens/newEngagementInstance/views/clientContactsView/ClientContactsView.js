import { memo } from 'react';

import CollapsibleFormSection from '../../../../common/collapsibleFormSection/CollapsibleFormSection';
import ButtonContainer from './components/ButtonContainer';
import CardList from './components/CardList';

let ClientContactsView = () => {
  // **********************************************************************
  // * constants

  // **********************************************************************
  // * component vars

  // **********************************************************************
  // * functions

  // **********************************************************************
  // * side effects

  // **********************************************************************
  // * render

  return (
    <div id="client-contacts-view">
      <CollapsibleFormSection title="Client Contacts">
        <div className="container-fluid">
          <ButtonContainer />
          <CardList />
        </div>
      </CollapsibleFormSection>
    </div>
  );
};

ClientContactsView = memo(ClientContactsView);
ClientContactsView.displayName = 'ClientContactsView';

export default ClientContactsView;
