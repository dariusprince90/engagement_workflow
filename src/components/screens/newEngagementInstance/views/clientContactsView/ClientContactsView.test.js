import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ClientContactsView from './ClientContactsView';

// **********************************************************************
// * constants

const testIds = {
  collapsibleFormSection: 'collapsible-form-section',
  buttonContainer: 'button-container',
  cardList: 'card-list'
};

// **********************************************************************
// * functions

const getComponentToRender = () => {
  return <ClientContactsView />;
};

// **********************************************************************
// * mock external dependencies

jest.mock('../../../../common/collapsibleFormSection/CollapsibleFormSection', () => ({
  __esModule: true,
  default: ({ title, children }) => {
    const props = { title, children };
    return <fake-collapsible-form-section {...props} data-testid={testIds.collapsibleFormSection} />;
  }
}));

jest.mock('./components/ButtonContainer', () => ({
  __esModule: true,
  default: () => <fake-button-container data-testIds={testIds.buttonContainer} />
}));

jest.mock('./components/CardList', () => ({
  __esModule: true,
  default: () => <fake-card-list data-testid={testIds.cardList} />
}));

// **********************************************************************
// * unit tests

describe('ClientContactsView', () => {
  // **********************************************************************
  // * setup

  // **********************************************************************
  // * tear-down

  // **********************************************************************
  // * execution

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponentToRender(), div);
  });

  it('renders the ButtonContainer above the CardList within the collapsible form section', () => {
    const tree = renderer.create(getComponentToRender()).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
