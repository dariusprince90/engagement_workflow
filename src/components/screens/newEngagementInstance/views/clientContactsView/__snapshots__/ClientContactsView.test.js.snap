// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientContactsView renders the ButtonContainer above the CardList within the collapsible form section 1`] = `
<div
  id="client-contacts-view"
>
  <fake-collapsible-form-section
    data-testid="collapsible-form-section"
    title="Client Contacts"
  >
    <div
      className="container-fluid"
    >
      <fake-button-container
        data-testIds="button-container"
      />
      <fake-card-list
        data-testid="card-list"
      />
    </div>
  </fake-collapsible-form-section>
</div>
`;
